src/fork/mod.rs

```diff
@@ -35,9 +35,10 @@ impl Fork {
                         -1 => Err(ForkError::Failure),
                         0 => {
                             let mut ptsname_buf = vec![0; MAX_PTS_NAME];
-                            if let Err(cause) = master.ptsname_r(&mut ptsname_buf) {
-                                return Err(ForkError::BadMaster(cause));
-                            }
+                            //if let Err(cause) = master.ptsname_r(&mut ptsname_buf) {
```

Comment:
if mac doesn't support ptsname_r, maybe we need to create the pty with openpty?

Also, just noticing this now, but this fork code is wrong. You aren't allowed to allocate in the child after a fork() call and this `vec!` constructor does so. We should probably fix that at some point.